src/popup/popup.js

```diff
@@ -286,18 +286,27 @@ elements.searchIcon.addEventListener('click', () => {
     });
   elements.clearSearchButton[0].classList.remove('display-none');
 });
-
+elements.modal.addEventListener('click', () => {
+  elements.modalCancel.click();
+});
 elements.clearSearchButton[0].addEventListener('click', () => {
-  // Restore Initial Content
-  elements.clearSearchButton[0].classList.add('display-none');
-  elements.inputField.value = '';
-  removeOldSearchResults();
-  removeLoadMoreButton(elements.loadMoreButtonWrapper);
-  elements.gridPrimary.setAttribute('style', 'position: relative; height: 0px;');
-  localStorage.clear();
-  restoreInitialContent('primary');
-
-  applyFilters();
+  const modalText = 'Do you really want to clear the search?';
+  function onModalConfirm() {
+    // Restore Initial Content
+    elements.clearSearchButton[0].classList.add('display-none');
+    elements.inputField.value = '';
+    removeOldSearchResults();
+    removeLoadMoreButton(elements.loadMoreButtonWrapper);
+    elements.gridPrimary.setAttribute('style', 'position: relative; height: 0px;');
+    localStorage.clear();
+    restoreInitialContent('primary');
+    applyFilters();
```

Comment:
Don't write this code again. Use the already defined functions for this.

Reply:
I am unable to get the defined function for this, can you please tell me which function and file have this code.

Reply:
I have checked but other functions are not in the same context. I have seen the same kind of code at several places in the extension but all of them has a difference of couple lines of code. What we can do is make a utility function which is also aware of the context (like `showNotification` or `addspinner` functions).

For this PR, you can skip this.